In [39]:
import pandas as pd, numpy as np, matplotlib.pyplot as plt, seaborn as sns
%matplotlib inline
     

In [40]:

WH=pd.read_excel('Working Hours.xlsx')
WH.head(5)
     

Date                 Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11    9:30 am- 10:30 am    1.0    1.0    NaN    NaN     NaN   
1 2022-04-11   10:30 am- 11:30 am    1.0    1.0    NaN    NaN     NaN   
2 2022-04-12    9:30 am- 10:30 am    1.0    1.0    NaN    NaN     NaN   
3 2022-04-12  11:30 am - 12:30 pm    1.0    1.0    NaN    NaN     NaN   
4 2022-04-13    9:30 am- 10:30 am    NaN    1.0    NaN    NaN     NaN   

  Mode of class  
0        Online  
1        Online  
2        Online  
3        Online  
4        Online

In [41]:

WH.isna().sum()

Date              0
Time              0
Jikky            51
Kayis            72
Sinny            68
Criji            74
Johnan           82
Mode of class     0
dtype: int64

# Data Cleaning

In [42]:

WH =WH.fillna(0)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  
0        Online  
1        Online

In [79]:
WH.isna().sum()

Date               0
Time               0
Jikky              0
Kayis              0
Sinny              0
Criji              0
Johnan             0
Mode of class      0
Month              0
year               0
Day                0
Salary_1_jikky     0
Salary_1_Kayis     0
Salary_1_Sinny     0
Salary_1_Criji     0
Salary_1_Johnan    0
Salary_2_jikky     0
Salary_2_Kayis     0
Salary_2_Sinny     0
Salary_2_Criji     0
Salary_2_Johnan    0
dtype: int64

In [43]:
WH.shape

(91, 8)

In [44]:
WH.columns

Index(['Date', 'Time', 'Jikky', 'Kayis', 'Sinny', 'Criji', 'Johnan',
       'Mode of class'],
      dtype='object')

In [45]:
WH.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91 entries, 0 to 90
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Date           91 non-null     datetime64[ns]
 1   Time           91 non-null     object        
 2   Jikky          91 non-null     float64       
 3   Kayis          91 non-null     float64       
 4   Sinny          91 non-null     float64       
 5   Criji          91 non-null     float64       
 6   Johnan         91 non-null     float64       
 7   Mode of class  91 non-null     object        
dtypes: datetime64[ns](1), float64(5), object(2)
memory usage: 5.8+ KB


### Splitting date

In [46]:
WH['Month'] = pd.to_datetime(WH['Date']).dt.month
WH['year'] = pd.to_datetime(WH['Date']).dt.year
WH['Day'] = pd.to_datetime(WH['Date']).dt.day
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  
0        Online      4  2022   11  
1        Online      4  2022   11

# Data Analysis

### 1. find who had taken more class

In [47]:
WH[['Jikky', 'Kayis', 'Sinny', 'Criji', 'Johnan']].sum()

Jikky     71.2
Kayis     29.6
Sinny     54.5
Criji     43.0
Johnan    23.0
dtype: float64

### 2.Find the sum of Hours of each mentors

In [48]:
WH.groupby('Mode of class').sum()

Jikky  Kayis  Sinny  Criji  Johnan  Month    year   Day
Mode of class                                                         
Online          57.2   22.6   30.5   23.0    13.5    406  161760  1440
Physical        14.0    7.0   24.0   20.0     9.5     61   22242   195

In [49]:
moc=WH['Mode of class'].value_counts()
moc

Online      80
Physical    11
Name: Mode of class, dtype: int64

### 3.Find how many physical classes had taken by each mentors through out each month

In [50]:
WH.groupby('Mode of class')[['Jikky','Kayis','Sinny','Criji','Johnan']].sum()

Jikky  Kayis  Sinny  Criji  Johnan
Mode of class                                    
Online          57.2   22.6   30.5   23.0    13.5
Physical        14.0    7.0   24.0   20.0     9.5

In [51]:

WH.groupby('Month')[['Jikky','Kayis','Sinny','Criji','Johnan']].sum()

Jikky  Kayis  Sinny  Criji  Johnan
Month                                    
4       11.5   14.5    9.5    1.0     0.0
5       22.7   15.1   22.0   21.0    23.0
6       37.0    0.0   23.0   21.0     0.0

In [52]:
jm=pd.crosstab(WH['Johnan'],WH['Mode of class']).sum()
jm

Mode of class
Online      80
Physical    11
dtype: int64

In [53]:

km=pd.crosstab(WH['Kayis'],WH['Mode of class']).sum()
km

Mode of class
Online      80
Physical    11
dtype: int64

In [54]:
sm=pd.crosstab(WH['Sinny'],WH['Mode of class']).sum()
sm
     

Mode of class
Online      80
Physical    11
dtype: int64

In [55]:
WH.groupby('Month')['Mode of class'].value_counts()

Month  Mode of class
4      Online           25
5      Online           24
       Physical          5
6      Online           31
       Physical          6
Name: Mode of class, dtype: int64

In [56]:
WH.groupby('Month')[['Mode of class']].count()     

Mode of class
Month               
4                 25
5                 29
6                 37

### 4.Find by which time hour, most of the class had taken in month June

In [57]:
WH['Time'].mode()

0    9:30 am - 11:30 am
Name: Time, dtype: object

In [58]:
WH.groupby('Month')['Time'].value_counts()

Month  Time               
4      9:30 am- 10:30 am      12
       11:30 am - 12:30 pm     8
       9:30 am - 11:30 am      3
       10:30 am- 11:30 am      1
       11:30 am -  1:00 pm     1
5      9:30 am - 11:30 am     17
       1:30 pm- 3:30 pm        3
       9:30am-12:00am          2
       9:30am:4:30pm           2
       2: 30PM - 3:50 PM       1
       3: 00PM - 3:30 PM       1
       9:30 am - 11:40 am      1
       9:30am-1 PM             1
       9:30am-3:30pm           1
6      2:30 Pm-4:30pm         14
       11:30 am- 1:30 Pm       9
       9:30 am - 11:30 am      9
       10:00 Am-3:30 Pm        1
       10:00 am - 11:30 am     1
       12:15 pm -12:45 pm      1
       2:30 pm -4:00 pm        1
       9:30am-3:30pm           1
Name: Time, dtype: int64

In [59]:
WH.groupby('Month')[['Time']].count()

Time
Month      
4        25
5        29
6        37

In [60]:
WH.describe()

Jikky      Kayis      Sinny      Criji     Johnan      Month  \
count  91.000000  91.000000  91.000000  91.000000  91.000000  91.000000   
mean    0.782418   0.325275   0.598901   0.472527   0.252747   5.131868   
std     1.116701   0.888393   1.236527   1.184067   0.940962   0.819333   
min     0.000000   0.000000   0.000000   0.000000   0.000000   4.000000   
25%     0.000000   0.000000   0.000000   0.000000   0.000000   4.000000   
50%     0.000000   0.000000   0.000000   0.000000   0.000000   5.000000   
75%     2.000000   0.000000   0.500000   0.000000   0.000000   6.000000   
max     7.000000   7.000000   7.000000   7.000000   7.000000   6.000000   

         year        Day  
count    91.0  91.000000  
mean   2022.0  17.967033  
std       0.0   8.430699  
min    2022.0   1.000000  
25%    2022.0  11.500000  
50%    2022.0  20.000000  
75%    2022.0  24.500000  
max    2022.0  31.000000

### 5.If the the payment is 100 for one hour , then how much salary

In [61]:
WH['Salary_1_jikky']=(WH['Jikky']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  
0        Online      4  2022   11           100.0  
1        Online      4  2022   11           100.0

In [62]:
WH['Salary_1_Kayis']=(WH['Kayis']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  
0        Online      4  2022   11           100.0           100.0  
1        Online      4  2022   11           100.0           100.0

In [63]:
WH['Salary_1_Sinny']=(WH['Sinny']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022   11           100.0           100.0   
1        Online      4  2022   11           100.0           100.0   

   Salary_1_Sinny  
0             0.0  
1             0.0

In [64]:
WH['Salary_1_Criji']=(WH['Criji']*100)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022   11           100.0           100.0   
1        Online      4  2022   11           100.0           100.0   

   Salary_1_Sinny  Salary_1_Criji  
0             0.0             0.0  
1             0.0             0.0

In [65]:
WH['Salary_1_Johnan']=(WH['Johnan']*100)
WH.head(2)
     

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022   11           100.0           100.0   
1        Online      4  2022   11           100.0           100.0   

   Salary_1_Sinny  Salary_1_Criji  Salary_1_Johnan  
0             0.0             0.0              0.0  
1             0.0             0.0              0.0

### 6.If the the payment for one hour is 150 means how much salary did they got in each month

In [66]:

WH['Salary_2_jikky']=(WH['Jikky']*150)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022   11           100.0           100.0   
1        Online      4  2022   11           100.0           100.0   

   Salary_1_Sinny  Salary_1_Criji  Salary_1_Johnan  Salary_2_jikky  
0             0.0             0.0              0.0           150.0  
1             0.0             0.0              0.0           150.0

In [67]:
WH['Salary_2_Kayis']=(WH['Kayis']*150)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022   11           100.0           100.0   
1        Online      4  2022   11           100.0           100.0   

   Salary_1_Sinny  Salary_1_Criji  Salary_1_Johnan  Salary_2_jikky  \
0             0.0             0.0              0.0           150.0   
1             0.0             0.0              0.0           150.0   

   Salary_2_Kayis  
0           150.0  
1           150.0

In [68]:

WH['Salary_2_Sinny']=(WH['Sinny']*150)
WH.head(2)
     

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022   11           100.0           100.0   
1        Online      4  2022   11           100.0           100.0   

   Salary_1_Sinny  Salary_1_Criji  Salary_1_Johnan  Salary_2_jikky  \
0             0.0             0.0              0.0           150.0   
1             0.0             0.0              0.0           150.0   

   Salary_2_Kayis  Salary_2_Sinny  
0           150.0             0.0  
1           150.0             0.0

In [69]:

WH['Salary_2_Criji']=(WH['Criji']*150)
WH.head(2)
     

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  Day  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022   11           100.0           100.0   
1        Online      4  2022   11           100.0           100.0   

   Salary_1_Sinny  Salary_1_Criji  Salary_1_Johnan  Salary_2_jikky  \
0             0.0             0.0              0.0           150.0   
1             0.0             0.0              0.0           150.0   

   Salary_2_Kayis  Salary_2_Sinny  Salary_2_Criji  
0           150.0             0.0             0.0  
1           150.0             0.0             0.0

In [70]:

WH['Salary_2_Johnan']=(WH['Johnan']*150)
WH.head(2)

Date                Time  Jikky  Kayis  Sinny  Criji  Johnan  \
0 2022-04-11   9:30 am- 10:30 am    1.0    1.0    0.0    0.0     0.0   
1 2022-04-11  10:30 am- 11:30 am    1.0    1.0    0.0    0.0     0.0   

  Mode of class  Month  year  ...  Salary_1_jikky  Salary_1_Kayis  \
0        Online      4  2022  ...           100.0           100.0   
1        Online      4  2022  ...           100.0           100.0   

   Salary_1_Sinny  Salary_1_Criji  Salary_1_Johnan  Salary_2_jikky  \
0             0.0             0.0              0.0           150.0   
1             0.0             0.0              0.0           150.0   

   Salary_2_Kayis  Salary_2_Sinny  Salary_2_Criji  Salary_2_Johnan  
0           150.0             0.0             0.0              0.0  
1           150.0             0.0             0.0              0.0  

[2 rows x 21 columns]

### 7.Find by which mode of class, most of the time class has taken in month June

In [71]:
tm=pd.crosstab(WH['Month'],WH['Mode of class'])
tm
     

Mode of class  Online  Physical
Month                          
4                  25         0
5                  24         5
6                  31         6

In [72]:
tm.mode()

Mode of class  Online  Physical
0                  24         0
1                  25         5
2                  31         6

In [73]:
WH.columns

Index(['Date', 'Time', 'Jikky', 'Kayis', 'Sinny', 'Criji', 'Johnan',
       'Mode of class', 'Month', 'year', 'Day', 'Salary_1_jikky',
       'Salary_1_Kayis', 'Salary_1_Sinny', 'Salary_1_Criji', 'Salary_1_Johnan',
       'Salary_2_jikky', 'Salary_2_Kayis', 'Salary_2_Sinny', 'Salary_2_Criji',
       'Salary_2_Johnan'],
      dtype='object')

### 8.Create a dataframe using the data you got with salary of each month of each mentors

In [74]:
WH[['Salary_1_jikky', 'Salary_1_Kayis','Salary_1_Sinny','Salary_1_Criji','Salary_1_Johnan']].sum()

Salary_1_jikky     7120.0
Salary_1_Kayis     2960.0
Salary_1_Sinny     5450.0
Salary_1_Criji     4300.0
Salary_1_Johnan    2300.0
dtype: float64

In [75]:
salary1 = WH.groupby('Month')[['Salary_1_jikky', 'Salary_1_Kayis','Salary_1_Sinny','Salary_1_Criji','Salary_1_Johnan']].sum()
salary1

Salary_1_jikky  Salary_1_Kayis  Salary_1_Sinny  Salary_1_Criji  \
Month                                                                   
4              1150.0          1450.0           950.0           100.0   
5              2270.0          1510.0          2200.0          2100.0   
6              3700.0             0.0          2300.0          2100.0   

       Salary_1_Johnan  
Month                   
4                  0.0  
5               2300.0  
6                  0.0

In [76]:
#if they get 150 rs in per hour

WH[['Salary_2_jikky','Salary_2_Kayis', 'Salary_2_Sinny','Salary_2_Criji', 'Salary_2_Johnan']].sum()

Salary_2_jikky     10680.0
Salary_2_Kayis      4440.0
Salary_2_Sinny      8175.0
Salary_2_Criji      6450.0
Salary_2_Johnan     3450.0
dtype: float64

In [77]:
salary2 = WH.groupby('Month')[['Salary_2_jikky','Salary_2_Kayis', 'Salary_2_Sinny','Salary_2_Criji', 'Salary_2_Johnan']].sum()
salary2

Salary_2_jikky  Salary_2_Kayis  Salary_2_Sinny  Salary_2_Criji  \
Month                                                                   
4              1725.0          2175.0          1425.0           150.0   
5              3405.0          2265.0          3300.0          3150.0   
6              5550.0             0.0          3450.0          3150.0   

       Salary_2_Johnan  
Month                   
4                  0.0  
5               3450.0  
6                  0.0

### 9.who earned more salary if the payment for one hour is 100.

In [80]:
WH[['Salary_1_jikky', 'Salary_1_Kayis','Salary_1_Sinny','Salary_1_Criji','Salary_1_Johnan']].max()

Salary_1_jikky     700.0
Salary_1_Kayis     700.0
Salary_1_Sinny     700.0
Salary_1_Criji     700.0
Salary_1_Johnan    700.0
dtype: float64

# Conclusion

91 rows and 8 columns contained in this table.

Fill null is cleaned that it replaced 0 from nan

The date split into month, day and year as they added in each column.

Jikky took more class at the total of 71.2

The mentors have taken online class than physical.

Jikky mostly took the online class

Sinny mostly took the physical class

Kayis has took more hours of working in April, Kayis earned the most salary in April

Johnan has took more hours of working in May, Johnan earned a higher salary in May

Jikky has mostly took more hours of working in June, Jikky earned the most salary in June

Jikky is working more time.

More physical classes had taken in June.

The mentors mostly took the class from 9.30am to 10.30am in April

The mentors mostly took the class from 9.30am to 11.30am in May.

The mentors mostly took the class from 2.30pm to 4.30pm in June